packages/lodestar-validator/test/unit/services/block.test.ts

```diff
@@ -35,51 +40,41 @@ describe("block proposing service", function () {
       new InvalidBlockError({code: InvalidBlockErrorCode.DOUBLE_BLOCK_PROPOSAL} as any)
     );
 
-    const service = new BlockProposingService(
-      config,
-      [Keypair.generate()],
-      rpcClientStub,
-      slashingProtectionStub,
-      logger
-    );
+    const service = new BlockProposingService(config, [keypair], rpcClientStub, slashingProtectionStub, logger);
     const result = await service.createAndPublishBlock(0, 1, generateFork(), ZERO_HASH);
     expect(result).to.be.null;
   });
 
   it("should produce correct block - last signed is null", async function () {
+    const privateKey = bls.PrivateKey.fromBytes(toBufferBE(BigInt(98), 32));
+    const publicKey = privateKey.toPublicKey();
+    const keypair = {privateKey, publicKey};
+
     const slot = 2;
     rpcClientStub.beacon.blocks.publishBlock = sandbox.stub();
     rpcClientStub.validator.produceBlock.resolves(generateEmptyBlock());
 
     slashingProtectionStub.checkAndInsertBlockProposal.resolves();
 
-    const service = new BlockProposingService(
-      config,
-      [Keypair.generate()],
-      rpcClientStub,
-      slashingProtectionStub,
-      logger
-    );
+    const service = new BlockProposingService(config, [keypair], rpcClientStub, slashingProtectionStub, logger);
     const result = await service.createAndPublishBlock(0, slot, generateFork(), ZERO_HASH);
     expect(result).to.not.be.null;
     expect(rpcClientStub.beacon.blocks.publishBlock.calledOnce).to.be.true;
   });
 
   it("should produce correct block - last signed in previous epoch", async function () {
+    const privateKey = bls.PrivateKey.fromBytes(toBufferBE(BigInt(98), 32));
+    const publicKey = privateKey.toPublicKey();
+    const keypair = {privateKey, publicKey};
```

Comment:
These three lines are repeated a lot. Probably it's not necessary to pass around both the privateKey and the publicKey, only the privateKey. If it's out the scope of this PR add a note on issue as todo

Reply:
seems like a bigger refactor of the validator